src/psi/j/executors/flux.py

```diff
@@ -0,0 +1,278 @@
+"""This module contains the Flux :class:`~psi.j.JobExecutor`."""
+
+from __future__ import annotations
+
+import time
+import logging
+import copy
+import json
+from functools import partial
+
+from typing import Any, Optional, Dict
+
+from distutils.version import StrictVersion
+
+from psi.j import InvalidJobException
+from psi.j import Job, JobExecutorConfig, JobState, JobStatus, JobAttributes
+from psi.j import JobExecutor, ResourceSpecV1
+
+logger = logging.getLogger(__name__)
+
+
+class FluxJobExecutor(JobExecutor):
+    """
+    A job executor that runs jobs via Flux.
+    """
+
+    import radical.utils as _ru
+    import flux as _flux
+
+    _NAME_ = 'flux'
+    _VERSION_ = StrictVersion('0.0.1')
+
+    # map flux states to jpsi states.
+    _state_map = {'NEW': JobState.QUEUED,
+                  'DEPEND': None,
+                  'SCHED': None,
+                  'RUN': JobState.ACTIVE,
+                  'CLEANUP': None,
+                  'INACTIVE': JobState.COMPLETED}
+    _event_map = {'submit': JobState.QUEUED,
+                  # 'alloc': None,
+                  'start': JobState.ACTIVE,
+                  'finish': JobState.COMPLETED,
+                  # 'release': None,
+                  # 'free': None,
+                  # 'clean': None,
+                  'exception': JobState.FAILED}
+
+    def __init__(self, url: Optional[str] = None,
+                 config: Optional[JobExecutorConfig] = None) -> None:
+        """
+        Initializes a `FluxJobExecutor`.
+
+        :param url: Not used, but required by the spec for automatic initialization.
+        :param config: The `FluxJobExecutor` does not have any configuration options.
+        :type config: psi.j.JobExecutorConfig
+        """
+        # TODO: url is not passed
+        # if not url.startswith('flux://'):
```

Comment:
Puh, that's a good question!  In RP we work around this by translating `flux://foo:123/bar` to `ssh://foo:123/bar` befor handing this to flux - but that only works because we always use the `ssh` channel (the RP executor does not necessarily live on the same node as the thing that started flux).  Supporting both protocols makes sense for J/PSI, but then the simple replacement is not sufficient.

This might be a point in favor of, IIUC, the mechanism @hategan proposes: select the backend during construction, and specify the endpoint URL via some configuration.  I am not a fan of packing too many things into configs, but that would work here.